cep-9999.md

```diff
@@ -0,0 +1,64 @@
+# CEP ???? - Source provenance metadata
+
+<table>
+<tr><td> Title </td><td> CEP ???? -  Source provenance metadata </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Mar 10, 2025</td></tr>
+<tr><td> Updated </td><td> Mar 10, 2025</td></tr>
+<tr><td> Discussion </td><td> N/A </td></tr>
+<tr><td> Implementation </td><td> N/A </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to standardize how the conda ecosystem can leverage free-form
+artifact metadata to annotate source and ecosystem provenance of the distributed
+artifacts. 
+
+We propose using the `about` field in the recipe metadata to implement
+a `purl` field where one or more [Package URLs](https://github.com/package-url/purl-spec)
+(PURLs) can be specified.
+
+## Motivation
+
+Provenance metadata is useful to assess from which sources an artifact was built.
+conda recipes already specify the necessary sources in their `source` section, but this field
+is mostly concerned with source download and preparation for the build. In
+most cases, it will contain one or more URLs or source-controlled repositories. These
+URLs cannot be guaranteed to uniquely identify a project since they may change, and their
+syntax might evolve across releases (e.g. different versioning schemes).
+
+Instead we propose adding a new field that _uniquely_ identifies the project being built.
+The Package URL project is chosen for this endeavour. This is speciallyuseful
+for efforts like dependency mapping across ecosystems. See
+[PEP 725](https://peps.python.org/pep-0725/) for practical applications in the
+context of PyPI <> conda interoperability.
+
+Example for `django`:
+
+```yaml
+# meta.yaml
+{% set version = "1.10.1" %}
+package: 
+  name: django
+  version: {{ version }}
+# ...
+about:
+  # ...
+  purls:
+    - pkg:pypi/django@{{ version }}
```

Comment:
Should we consider making the `purls` a mapping, rather than a simple list? This would make the field more easily interpretable and would give us more flexibility to further extend it.

E.g.,
```
purls:
  upstream:
    - pkg:pypi/foo
  vendors:
    - pkg:github/bar
    - pkg:github/baz
```

Reply:
A free-form mapping or a mapping with well defined keys? If the latter, which ones? I don't want to end up in SBOM-like territory.